Sources/Host/POSIX/File.cpp

```diff
@@ -23,5 +24,31 @@ File::File(std::string const &path, uint32_t flags, uint32_t mode) {
 }
 
 File::~File() { ::close(_fd); }
+
+ErrorCode File::pread(std::string &buf, uint64_t &count, uint64_t offset) {
+  if (_fd < 0) {
+    return _lastError = kErrorInvalidHandle;
+  }
+
+  if (offset > std::numeric_limits<off_t>::max()) {
+    return _lastError = kErrorInvalidArgument;
+  }
+
+  off_t offArg = static_cast<off_t>(offset);
+  size_t countArg = static_cast<size_t>(count);
```

Comment:
For lines 37 and 38, you should use `auto` instead of repeating the type both in the variable declaration and the `static_cast` parameter.